core/src/main/java/hudson/model/RunMap.java

```diff
@@ -228,9 +228,11 @@ protected R retrieve(File d) throws IOException {
                 }
                 return b;
             } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "could not load " + d, e);
+                throw new IOException();
```

Comment:
IMO you could re-throw the existing exception or add it as a nested one.
E.g. <code>throw new IOException("could not load " + d, e)</code>